### ISSUE TITLE
Testnet Encrypted ExtKey

### ISSUE THREAD

**Issue by Chaz27** [NicolasDorier/NBitcoin#58]
Hi, I am having trouble generating an ExtKey, encrypting it, then restoring it - specifically with Testnet.  I can successfully get it to work with Main as shown here:

````C#
Network network = Network.Main;

ExtKey extKey = new ExtKey();
string extPubKey = extKey.Neuter().GetWif(network).ToWif();
// xpub661MyMwAqRbcFbAZaNJrgn2fBgjkFCWoBjUAnMisnMtt4qvtqNNESfAwusT31U4Eb43z8YdA36zrpf7vcsKvH5yi9vaYzMzftHZC7gb44n5
string extPrivateKey = extKey.GetWif(network).ToWif();
// xprv9s21ZrQH143K3766ULmrKe5vdeuFqjnwpWYZyyKGE2MuC3bkHq3ytrrU4biSM1nxfUDkXo4gopNSCkG4SnLf5TGd5m4mJ1FCqKHphSjUTa6
string encryptedPrivateKey = extKey.Key.GetEncryptedBitcoinSecret("abc123", network).ToWif();
// 6PYW7angZfaQ3SZdhDm8CsmzM9Gaue7LYxXwLCFXqgLEa96CeKTQMeNBfH

Key key = Key.Parse(encryptedPrivateKey, "abc123", network);
ExtKey extKey2 = new ExtKey(ExtPubKey.Parse(extPubKey), key);
string extPrivateKey2 = extKey2.GetWif(network).ToWif();
// xprv9s21ZrQH143K3766ULmrKe5vdeuFqjnwpWYZyyKGE2MuC3bkHq3ytrrU4biSM1nxfUDkXo4gopNSCkG4SnLf5TGd5m4mJ1FCqKHphSjUTa6
````

As you can see the private key's are the same.  However when changing the network to `Network.TestNet` I get the following error:

````
System.FormatException: Invalid network
at NBitcoin.Network.CreateFromBase58Data(String base58, Network expectedNetwork) in Network.cs:line 372
at NBitcoin.Network.CreateFromBase58Data[T](String base58, Network expectedNetwork) in Network.cs:line 381
at NBitcoin.Key.Parse(String wif, String password, Network network) in Key.cs:line 24
````

I'm not sure if I am doing something wrong in the above code, please advise.

Cheers

**Comment by NicolasDorier** [comment]
The root of the problem is that from [BIP 38](https://github.com/bitcoin/bips/blob/master/bip-0038.mediawiki), there is no testnet base58 prefix specifically for testnet.
Thus, the network is not stored into the base58 string.

However, I admit such code should not break. I will update and release a new nuget version today fixing this problem. Thanks for reporting.

**Comment by NicolasDorier** [comment]
Fixed in f3dfe2dea17988717f161c34c340674114d177fa
Nuget NBitcoin updated, run,

> Update-Package NBitcoin

More [info](http://docs.nuget.org/docs/start-here/using-the-package-manager-console).

Keep in mind that the base58 of your encrypted key do not store the Network information. 
So if you generate an encrypted key on TestNet, and parse on MainNet you'll get "invalid password".

Reopen the issue if any remaining problem.

**Comment by Chaz27** [comment]
Thanks Nicolas, really enjoy using NBitcoin as well!
